app/controllers/api/v1/shipper/invoices_controller.rb

```diff
@@ -25,31 +26,33 @@ def show
   end
 
   def update
-    @user_invoice = @invoice.user_invoices.find_by_user_id current_user.id
-    if check_update_status? && InvoiceStatus.new(@invoice, @user_invoice,
-      params[:status]).shipper_update_status
+    if InvoiceStatus.new(@invoice, @user_invoice, params[:status],
+      current_user).shipper_update_status
       render json: {message: I18n.t("invoices.messages.update_success"),
         data: {invoice: @invoice}, code: 1}, status: 200
     else
-      render json: {message: I18n.t("invoices.messages.invoice_error_status"),
+      render json: {message: I18n.t("invoices.messages.update_fail"),
         data: {}, code: 0}, status: 200
     end
   end
 
   private
-  def check_update_status?
-    if params[:status] == "cancel" && !@invoice.finished?
-      true
-    elsif Invoice.statuses[params[:status]].pred == Invoice.statuses[@invoice.status]
-      true
-    else
-      false
+  def check_conditions_to_update_status?
+    @user_invoice = @invoice.user_invoices.find_by_user_id current_user.id
+    unless !@user_invoice.nil? &&
```

Comment:
anh nghĩ chỗ này dùng if sẽ tốt hơn unless, sau này sẽ rất khó hiểu vì unless and khó lắm